octobot_tentacles_manager/uploaders/s3_uploader.py

```diff
@@ -0,0 +1,96 @@
+#  Drakkar-Software OctoBot-Tentacles-Manager
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import os
+import aioboto3
+
+import octobot_tentacles_manager.uploaders.uploader as uploader
+
+
+class S3Uploader(uploader.Uploader):
+    ENV_S3_API_KEY = "S3_API_KEY"
+    ENV_S3_API_SECRET_KEY = "S3_API_SECRET_KEY"
+    ENV_S3_ENDPOINT_URL = "S3_ENDPOINT_URL"
+    ENV_S3_BUCKET_NAME = "S3_BUCKET_NAME"
+    ENV_S3_REGION_NAME = "S3_REGION_NAME"
+
+    S3_BOTO_KEY = "s3"
+
+    def __init__(self):
+        super().__init__()
+        self.s3_session: aioboto3.Session = aioboto3.Session()
+        self.s3_api_key: str = os.getenv(S3Uploader.ENV_S3_API_KEY, None)
+        self.s3_api_secret_key: str = os.getenv(S3Uploader.ENV_S3_API_SECRET_KEY, None)
+        self.s3_endpoint_url: str = os.getenv(S3Uploader.ENV_S3_ENDPOINT_URL, None)
+        self.s3_bucket_name: str = os.getenv(S3Uploader.ENV_S3_BUCKET_NAME, None)
+        self.s3_region_name: str = os.getenv(S3Uploader.ENV_S3_REGION_NAME, None)
+        if None in [self.s3_api_key, self.s3_api_secret_key, self.s3_endpoint_url,
+                    self.s3_bucket_name, self.s3_region_name]:
```

Comment:
It's a detail but creating a tuple instead of a lust here would be faster :)

Reply:
:+1: